File: C/problem3.c
#include <stdio.h>

int largestfactor(long long int num){
	int lastFactor;
	if(num%2==0){
		lastFactor=2;
		while(num%2==0)
			num/=2;
	}
	else
		lastFactor=1;
	int factor=3;
	while(num>1){
		if(num%factor==0){
			num/=factor;
			lastFactor=factor;
			while(num%factor==0)
				num/=factor;
		}
		factor=factor+2;
	}
	if(num==1)
		return lastFactor;
	else
		return num;
}
int main(){
	printf("Factor is: %d\n",largestfactor(600851475143));
}

File: Ruby/problem3.rb
def largestFactor(n)
	lastFactor=0
	if n%2==0 then
		lastFactor=2
		while n%2==0 do
			n/=2			
		end
	else
		lastFactor=1
	end
	factor=3
	while n>1 do
		if n%factor==0 then
			lastFactor=factor
			while n%factor==0 do
				n/=factor
			end
		end
		factor+=2
	end
	puts lastFactor
end